src/articles/2024-01-21_DDEV-for-js.md

```diff
@@ -0,0 +1,327 @@
+
+If you'd like to have actual running apps for both of these to see everything working, use each app's quickstart commands:
+
+```txt
+$ npm create keystone-app keystone
+```
+
+```txt
+$ npm create svelte@latest svelte
+```
+
+Both of these commands will end with instructions to start the applications, but we'll skip that for now. Our project structure should now look like this:
+
+```txt
+my-project
+|- keystone
+|- svelte
+```
+
+From `my-project`, run `ddev config`.
+
+```txt
+$ ddev config
+
+Creating a new DDEV project config in the current directory (~/my-project)
+Once completed, your configuration will be written to ~/my-project/.ddev/config.yaml
+
+Project name (my-project):
+
+The docroot is the directory from which your site is served.
+This is a relative path from your project root at /Users/andy/Sites/my-project
+You may leave this value blank if your site files are in the project root
+Docroot Location (current directory):
+
+Found a php codebase at /Users/andy/Sites/my-project.
+Project Type [backdrop, craftcms, django4, drupal10, drupal6, drupal7, drupal8, drupal9, laravel, magento, magento2, php, python, shopware6, silverstripe, typo3, wordpress] (php):
+
+Configuration complete. You may now run 'ddev start'.
+```
+
+First, DDEV will ask for your project name. This value will be used in the primary URL for the project, so make sure it follows the rules for [hostname syntax](https://en.wikipedia.org/wiki/Hostname#Syntax). Next, DDEV will attempt to identify what kind of project you have. Since DDEV was originally built with php-based CMSes in mind, the default project type is "php", meaning there's no special extra configuration happening. There is currently [an open issue](https://github.com/ddev/ddev/issues/5722) to consider renaming this to something more helpful like "generic" or "any".
+
+If you run `ddev start` now, you'll have a working `web` and `db` container!
+
+```txt
+$ ddev start
+
+Starting my-project...
+Network ddev-my-project_default created
+Building project images...
+Project images built in 0s.
+ Container ddev-my-project-db  Created
+ Container ddev-my-project-web  Created
+ Container ddev-my-project-web  Started
+ Container ddev-my-project-db  Started
+You have Mutagen enabled and your 'php' project type doesn't have `upload_dirs` set.
+For faster startup and less disk usage, set upload_dirs to where your user-generated files are stored.
+If this is intended you can disable this warning with `ddev config --disable-upload-dirs-warning`.
+Starting Mutagen sync process...
+Mutagen sync flush completed in 1s.
+For details on sync status 'ddev mutagen st my-project -l'
+Waiting for web/db containers to become ready: [web db]
+^[Starting ddev-router if necessary...
+ Container ddev-router  Running
+Waiting for additional project containers to become ready...
+All project containers are now ready.
+Successfully started my-project
+Project can be reached at https://my-project.ddev.site https://127.0.0.1:33125
```

Comment:
I wonder if it's worth pushing up the actual repo (with a name other than `my-project`), and immediately archiving it.

Reply:
This seems like a good idea. I'll try to take up @rfay on his offer of an on-call walkthrough and then add each step in its own commit.